packages/web-app-files/src/helpers/resources.ts

```diff
@@ -89,6 +89,9 @@ function addSharedWithToShares(shares) {
       previousShare?.file_source === share.file_source
     ) {
       if (ShareTypes.containsAnyValue(ShareTypes.authenticated, [parseInt(share.share_type)])) {
+        if (share.stime > previousShare.stime) {
+          previousShare.stime = share.stime
+        }
```

Comment:
Hmm I'm not the biggest fan of altering the `stime` of the share here. Can't we achieve the same by simply sorting the spaces before iterating them? Something like: `for (const share of shares.sort((a: any, b: any) => b.stime - a.stime))`

Reply:
Realistically it has to be this
```
  shares.sort((a: any, b: any) =>
    a.storageId === b.storageId && a.file_source === b.file_source ? b.stime - a.stime : 0
  )
```
Those shares are not grouped by file, so only sorting by stime would display duplicates. I'm not sure it's that performant using sort while the cycle has to run anyway, I would say it's less readable. 
Maybe the whole function might benefit from a little rewrite, group the individual shares by file, then sort inside with simpler comparison, it should be little more cleaner and readable

Reply:
> Maybe the whole function might benefit from a little rewrite, group the individual shares by file, then sort inside with simpler comparison, it should be little more cleaner and readable

I agree, but let's not make this part of the bug fix 🙂 Let's go with your proposed solution then.